### PR TITLE
ref(charts) Extract charts into a separate webpack chunk

### PR DESCRIPTION
**Pull request by markstory**
Code splitting echarts better lets us shave megabytes off of the discover and performance routes and allows code sharing between discover, alerts and performance views. There are still some echarts bits being loaded by dashboards for maps.

## Bundle comparison

Screen Shot 2020-08-24 at 9 16 43 PM

On the left is the *new* state, while the right has *master*. There are fewer echarts files bundled (less duplication) and less overall file size as well. The separate echarts bundle should do better with browser caching as it will change less frequently now as it only really contains echarts and zrender.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/components/charts/baseChart.tsx [hunk 1]
```diff
@@ -239,6 +242,30 @@ class BaseChart extends React.Component<Props> {
...
   };
 
+  state: State = {
+    chartDeps: undefined,
+  };
+
+  componentDidMount() {
+    this.loadEcharts();
+    this._isMounted = true;
+  }
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/charts/baseChart.tsx:252]: Shouldn't this be placed before the function `loadEcharts`?

**Review by priscilawebdev — APPROVED** [review]
Great stuff!! 👏

**Review by matejminar — APPROVED** [review]
Nice

**Review by billyvg — APPROVED** [review]
The trade-off here is that certain views will have a larger initial page load since the echarts bundle will include unused parts, however this is offset by being able to cache the echarts bundle (if we had a long lived cache).

I generally would prefer to let webpack do its thing, but having the echarts component libs live with its component counterparts as been a source of confusion and I think this should simplify things quite a bit. And if we had smarter caching, the echarts bundle would be a huge win as well.

**Review by k-fish — APPROVED** [review]
To Billy's point, if we know which page we are going to incur the most additional load on we can watch for an uptick in `transaction.op:pageload` duration for that transaction to confirm if it has a tangible impact on performance.

**Pull request merged**